Intersect.Client/Entities/Player.cs

```diff
@@ -2252,124 +2252,71 @@ public virtual void DrawGuildName(Color textColor, Color borderColor = null, Col
+            // Cache values available out of the loop.
+            Type[] ignoreEntityTypes = { typeof(Projectile), typeof(Resource), typeof(Event) };
+            var mapId = map.Id;
+            var mousePos = Graphics.ConvertToWorldPoint(Globals.InputManager.MousePosition);
+            bool myOverheadPref = Globals.Database.MyOverheadInfo,
+                npcOverheadPref = Globals.Database.NpcOverheadInfo,
+                playerOverheadPref = Globals.Database.PlayerOverheadInfo,
+                friendOverheadPref = Globals.Database.FriendOverheadInfo,
+                guildMemberOverheadPref = Globals.Database.GuildMemberOverheadInfo,
+                partyMemberOverheadPref = Globals.Database.PartyMemberOverheadInfo;
+
+            // Iterate over the entities separately.
+            foreach (var en in Globals.Entities)
             {
-                if (mousePos.X >= map.GetX() && mousePos.X <= map.GetX() + Options.MapWidth * Options.TileWidth)
+                if (en.Value == null || en.Value.MapId != mapId)
                 {
-                    if (mousePos.Y >= map.GetY() && mousePos.Y <= map.GetY() + Options.MapHeight * Options.TileHeight)
-                    {
-                        var mapId = map.Id;
-                        foreach (var en in Globals.Entities)
-                        {
-                            if (en.Value == null)
-                            {
-                                continue;
-                            }
-                            
-                            en.Value.IsHovered = false;
-
-                            var isPlayer = en.Value is Player;
-                            var isNpc = !isPlayer;
-                            var player = en.Value as Player;
-
-                            if (en.Value.MapId == mapId &&
-                                !en.Value.HideName &&
-                                (!en.Value.IsStealthed || isPlayer && Globals.Me.IsInMyParty(player)) &&
-                                en.Value.WorldPos.Contains(mousePos.X, mousePos.Y))
-                            {
-                                // We don't want to deal with these entities when hovering the cursor by them.
-                                var ignoreEntities = !(en.Value is Projectile || en.Value is Resource || en.Value is Event);
-
-                                if (ignoreEntities)
-                                {
-                                    // Who's who.
-                                    var isMe = isPlayer && player.Id == Globals.Me.Id;
-                                    var isOtherPlayer = isPlayer && !isMe;
-                                    var isFriend = isOtherPlayer && Globals.Me.IsFriend(player);
-                                    var isGuildMate = isOtherPlayer && Globals.Me.IsGuildMate(player);
-                                    var isPartyMate = isOtherPlayer && Globals.Me.IsInMyParty(player);
```

Comment:
This caching is horrible and unnecessary

Reply:
removed `isNpc` and `isNpc` since they were used only once. kept the rest since they are used 2 to 3 + times.

Reply:
got rid of the variables that were used only once and took another approach by moving these out so we don't iterate though these booleans on every foreach loop.